packages/css/search.css

```diff
@@ -1,170 +1,108 @@
-  font-size: 1.25rem;
-  padding: 0;
-  z-index: 2;
-}
-
-.ds-search--sm .ds-search__input {
-  --dsc-search-clear-button-size: var(--ds-sizing-4);
-
-  height: var(--ds-sizing-10);
-  padding: 0 var(--ds-spacing-3);
-  padding-right: 2.5em;
-}
-
-.ds-search--sm .ds-search__icon {
-  left: var(--ds-spacing-3);
-}
-
-.ds-search--md .ds-search__input {
-  --dsc-search-clear-button-size: var(--ds-sizing-6);
-
-  height: var(--ds-sizing-12);
-  padding: 0 var(--ds-spacing-4);
-  padding-right: 2.2em;
-}
-
-.ds-search--md .ds-search__icon {
-  left: var(--ds-spacing-4);
-}
+  /* The simple variant, when no submit button is present */
+  &:not(:has([type='submit'])) {
+    & input {
+      padding-left: 2.4em;
+    }
+
+    &::before {
+      content: '';
+      position: absolute;
+      left: var(--ds-sizing-2);
+      top: 50%;
+      transform: translateY(-50%);
+      height: var(--dsc-search-icon-size);
+      aspect-ratio: 1 / 1;
+      mask: var(--dsc-search-input-icon) center / contain no-repeat;
+      background: currentcolor;
+      z-index: 2;
+      pointer-events: none;
+    }
+  }
+  & input {
+    grid-column: 1 / 2;
+    grid-row: 1 / 2;
+
+    &:focus-visible {
+      z-index: 1;
+    }
+    &::-webkit-search-decoration,
+    &::-webkit-search-cancel-button {
+      appearance: none;
+    }
+  }
 
-.ds-search--lg .ds-search__input {
-  --dsc-search-clear-button-size: var(--ds-sizing-12);
+  &:has(input:placeholder-shown) [type='reset'] {
+    display: none;
+  }
 
-  height: var(--ds-sizing-14);
-  padding: 0 var(--ds-spacing-5);
-  padding-right: 2em;
-}
+  & [type='reset'] {
+    --dsc-button-padding: 0;
+
+    grid-column: 1 / 2;
+    grid-row: 1 / 2;
+    justify-self: end;
+    align-self: center;
+    margin-right: var(--ds-spacing-3);
+    border: none;
+    cursor: pointer;
+    height: var(--dsc-search-clear-button-size);
+    width: var(--dsc-search-clear-button-size);
+    border-radius: var(--ds-border-radius-default);
+    font-size: 0.5em;
+    z-index: 2;
+    position: relative;
+
+    &::before {
+      content: '';
+      position: absolute;
+      left: 50%;
+      top: 50%;
+      transform: translate(-50%, -50%);
+      height: var(--dsc-search-clear-icon-size);
+      aspect-ratio: 1 / 1;
+      mask: var(--dsc-search-clear-icon) center / contain no-repeat;
+      background: currentcolor;
+      z-index: 2;
+      pointer-events: none;
+    }
+  }
 
-.ds-search--lg .ds-search__icon {
-  left: var(--ds-spacing-5);
-}
+  & [type='submit'] {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
 
-.ds-search__input.ds-search__input--simple {
-  padding-left: 2.4em;
-}
+    &:focus-visible {
+      z-index: 1;
+    }
 
-@media (hover: hover) and (pointer: fine) {
-  .ds-search__input:not(:focus-visible, :disabled, [aria-disabled]):hover {
-    --dsc-search-input-border-color: var(--ds-color-accent-border-strong);
+    &:not(:focus-visible) {
+      border-left: 0;
```

Comment:
Why is this? :)

Reply:
We want the button to have a border left if it is focused, but not when it is not.
If we don't have this then we will get double borders between a secondary button and the input

Reply:
Right, but maybe we could rater just hide the border-right on the input if search-button exists? that border can be hidden regardless of focus, and you already have a selector for it :)

Reply:
We still need to border for the input if it is focused

Reply:
ah dang it :p